ginkgo-kubernetes-all.Jenkinsfile

```diff
@@ -56,12 +56,41 @@ pipeline {
                 }
             }
         }
+        stage('Non-release-k8s-versions') {
+            environment {
+                GOPATH="${WORKSPACE}"
+                TESTDIR="${WORKSPACE}/${PROJ_PATH}/test"
+            }
+            options {
+                timeout(time: 60, unit: 'MINUTES')
+            }
+            steps {
+                parallel(
+                    "K8s-1.10":{
+                        sh 'cd ${TESTDIR}; K8S_VERSION=1.10 ginkgo --focus=" K8s*" -v -noColor'
+                    },
+                    "K8s-1.11":{
+                        sh 'cd ${TESTDIR}; K8S_VERSION=1.11 ginkgo --focus=" K8s*" -v -noColor'
+                    },
+                )
+            }
+            post {
+                always {
+                    junit 'test/*.xml'
+                    sh 'cd test/; ./post_build_agent.sh || true'
+                    sh 'cd test/; ./archive_test_results.sh || true'
+                    archiveArtifacts artifacts: "test_results_${JOB_BASE_NAME}_${BUILD_NUMBER}.tar", allowEmptyArchive: true
+                }
+            }
+        }
     }
     post {
         always {
             cleanWs()
             sh "cd ${TESTDIR}; K8S_VERSION=1.7 vagrant destroy -f || true"
             sh "cd ${TESTDIR}; K8S_VERSION=1.9 vagrant destroy -f || true"
+            sh "cd ${TESTDIR}; K8S_VERSION=1.10 vagrant destroy -f || true"
+            sh "cd ${TESTDIR}; K8S_VERSION=1.11 vagrant destroy -f || true"
```

Comment:
Will having 4 sets of K8s VMs running on the host at once be feasible? If the stages are run in serial order (i.e., 1.7 and 1.9 then 1.10 and 1.11), I would delete 1.7 and 1.9 VMs in that stage's `post` section vs. waiting until the end of the entire job.

Reply:
Hi @ianvernon 

Servers are deleted in Ginkgo:
https://github.com/cilium/cilium/blob/master/test/test_suite_test.go#L194-L200

That lines are to make sure that no leftover VMS are in place, but Ginkgo uses to delete all ;-)